### PR TITLE
[SEPOLICY] Add support for the Camera Augmented Sensing Helper Server

### PR DESCRIPTION
**Pull request by kholk**
Needs context and permission to read input devices, search,
read and write its own sockets.

Also needs camera HAL to access its socket.

### CODE REVIEW

File: hal_camera_default.te [hunk 1]
```diff
@@ -18,6 +18,10 @@ allow hal_camera_default hal_power_default:unix_stream_socket connectto;
 allow hal_camera_default powerhal_socket:dir search;
 allow hal_camera_default powerhal_socket:sock_file write;
 
+allow hal_camera_default cashsvr:unix_stream_socket connectto;
+allow hal_camera_default cashsvr_socket:dir search;
+allow hal_camera_default cashsvr_socket:sock_file write;
```
Comment (luk1337) [inline hal_camera_default.te]: Use unix_socket_connect() macro.

File: cashsvr.te [hunk 2]
```diff
@@ -0,0 +1,7 @@
+type cashsvr, domain;
+type cashsvr_exec, exec_type, vendor_file_type, file_type;
+
+allow cashsvr cashsvr_socket:dir rw_dir_perms;
+allow cashsvr cashsvr_socket:sock_file create_file_perms;
+
+init_daemon_domain(cashsvr)
```
Comment (luk1337) [inline cashsvr.te]: Move it after type definitions.

File: hal_camera_default.te (hunk 1 above)
Comment (kholk) [reply hal_camera_default.te]: Perhaps everything else using sockets has to be fixed as well, not only my addition.
In this case, I'd rather say that it's better to just merge it as it is and then do a commit that does the cosmetic change that you are requesting on this plus all other non-CASH-related occurrences.

**Comment by jerpelea** [comment]
I agree
please make the cleanup after this merge so that we have a split between additions and cleanup

**Comment by jerpelea** [comment]
please do the suggested changes without cleanup

**Pull request closed without merging**